### PR TITLE
add mpi conceptual guide

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##

Add conceptual guide and image

@siddvenk

### CODE REVIEW

File: serving/docs/lmi/conceptual_guide/lmi_engine.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+serving.properties
+
+```
+engine=Python
+```
+
+Environment variable
+
+```
+SERVING_LOAD_MODELS=test::Python=/opt/ml/model
```
Comment (frankfliu) [inline serving/docs/lmi/conceptual_guide/lmi_engine.md:39]: I don't think we want to promote this usage. We have to explain what `test` mean here.

In most cases, we are able to detect Python engine. We don't have to explicitly enable python engine. If really this is needed, we can introduce `SERVING_ENGINE` env var.

For MPI mode, we already have an env var: 'OPTION_MPI_MODE=true`

Comment (siddvenk) [reply serving/docs/lmi/conceptual_guide/lmi_engine.md:39]: For v9 we can rely just on HF_MODEL_ID (and optionally OPTION_ROLLING_BATCH if user wants to specify). But for v8, the auto Python engine mode logic doesn't really work if user specifies rolling batch but not engine.

Comment (siddvenk) [reply serving/docs/lmi/conceptual_guide/lmi_engine.md:39]: I'm wrong. In V8 the following are sufficient:
OPTION_MODEL_ID/HF_MODEL_ID
OPTION_ROLLING_BATCH
OPTION_MPI_MODE

we don't need SERVING_LOAD_MODELS even in v8

File: serving/docs/lmi/conceptual_guide/lmi_engine.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+![python image](../imgs/python_mode.jpg)
+
+DJLServing could operate with Python with our Python Engine. In Python operating mode, we will spin up a python process
+from the system environment and allocate Accelerators(CPU/Neuron) for each processes through `CUDA_VISIBLE_DEVICES`. During auto-scaling mode,
+DJLServing could manage workers' Accelerators allocation and spin up process with different Accelerators (CPU/Neuron).
+Under python Engine mode, DJLServing will establish socket connection and talk to the python process.
+
+### Enablement
+
+You can use the following ways to enable MPI Engine:
```
Comment (siddvenk) [inline serving/docs/lmi/conceptual_guide/lmi_engine.md:28]: MPI -> Python

File: serving/docs/lmi/conceptual_guide/lmi_engine.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+We use python mode as long as you specify `option.model_id`.
+
+## MPI Engine operating in LMI (DJLServing)
+
+![mpi image](../imgs/mpi_mode.jpg)
+
+MPI in general means "Multi-Process-Interface". In LMI domain, you could also read as "Multi-Process-Inference".
+DJLServing internally will use `mpirun` to spin up multiple processes depends on the setup.
+The number of process for LLM applications following `tensor_parallel_degree x pipline_parallel_degree`.
```
Comment (siddvenk) [inline serving/docs/lmi/conceptual_guide/lmi_engine.md:46]: should we remove pipeline_parallel_degree? We don't have any examples of using both and don't really talk about it anywhere else. Might be confusing to users even though it's accurate.

File: serving/docs/lmi/conceptual_guide/lmi_engine.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+- TransformerNeuronX (Python): The Neuron backend engine, internally it will use multi-threading to run with Neuron cores.
+
+In the next section, we will introduce a detailed breakdown on how we run those backends.
+
+## Python Engine operating in LMI (DJLServing)
+
+![python image](../imgs/python_mode.jpg)
+
+DJLServing could operate with Python with our Python Engine. In Python operating mode, we will spin up a python process
+from the system environment and allocate Accelerators(CPU/Neuron) for each processes through `CUDA_VISIBLE_DEVICES`. During auto-scaling mode,
```
Comment (siddvenk) [inline serving/docs/lmi/conceptual_guide/lmi_engine.md:22]: CPU -> GPU

File: serving/docs/lmi/conceptual_guide/lmi_engine.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+DJLServing could manage workers' Accelerators allocation and spin up process with different Accelerators (CPU/Neuron).
```
Comment (siddvenk) [inline serving/docs/lmi/conceptual_guide/lmi_engine.md:23]: CPU -> GPU

**Review by siddvenk — APPROVED** [review]

**Pull request merged**